boards/shields/ssd1306/Kconfig.defconfig

```diff
@@ -1,13 +1,24 @@
 # Copyright (c) 2019 Linaro Limited
 # SPDX-License-Identifier: Apache-2.0
 
-if SHIELD_SSD1306_128X64 || SHIELD_SSD1306_128X32 || SHIELD_SH1106_128X64
-
 if DISPLAY
 
+if SHIELD_SSD1306_128X64 || SHIELD_SSD1306_128X64_SPI || SHIELD_SSD1306_128X32 || SHIELD_SH1106_128X64
```

Comment:
Can you explain why you're inverting the lines here ?

Reply:
Basic idea was to have a hierarchical ordering from the most generic (DISPLAY) down to the actual types and their configurations (SPI or I2C).

Reply:
> Basic idea was to have a hierarchical ordering from the most generic (DISPLAY) down to the actual types and their configurations (SPI or I2C).

Ok, I agree on the general idea. Though in this concept, I think the highest prio would be: hardware is present (so one of `SHIELD_SSD1306_FOO` is true). So nothing in this file should be treated if this condition doesn't apply.
This is also quite easy to explain and apply. 
Last, this change breaks consistency with other shields, so it would have to be done in a dedicated PR, and take all shields into account.